### ISSUE TITLE
HowTo: Simplish SSH Tunnel

### ISSUE THREAD

**Issue by jmdobry** [mscdex/ssh2#262]
I'm trying to open an ssh tunnel in order to access a mysql database on a remote server, but I haven't been able to figure it out so far. How would I do `ssh -NC user@my.server.com -L 3306:127.0.0.1:3306` (uses my private key) with ssh2? (Would be good to show equivalent `ssh` commands for the examples.)

**Comment by mscdex** [comment]
First of all, compression is not yet supported. Eventually it will probably be implemented synchronously (which means node v0.12+ or io.js) because that'll be the easiest way.

However, to forward location connections to the remote server without compression, you can do something like:

```javascript
var net = require('net');
var SSH2Client = require('ssh2').Client;

var ssh = new SSH2Client();
ssh.on('ready', function() {
  console.log('SSH connection ready ...');
  net.createServer(function(sock) {
    ssh.forwardOut(sock.remoteAddress,
                   sock.remotePort,
                   '127.0.0.1',
                   3306,
                   function(err, stream) {
      if (err) return sock.end();
      sock.pipe(stream).pipe(sock);
    });
  }).listen(3306, function() {
    console.log('Listening on 3306 for connections to forward');
  });
}).connect({
  host: 'my.server.com',
  username: 'user',
  // assuming you are using an ssh-agent, otherwise change to whatever authentication mechanism
  agent: process.env.SSH_AUTH_SOCK
});
```

Obviously this can be improved upon by making things more reliable (e.g. handling the ssh connection getting severed, etc.), but hopefully it will guide you in the right direction.

**Comment by jmdobry** [comment]
Awesome thanks!

p.s. I didn't see a link to a mailing list or anything, otherwise I would have asked there.
